projects/igniteui-angular/src/lib/directives/mask/mask.directive.ts

```diff
@@ -274,12 +277,19 @@ export class IgxMaskDirective implements OnInit, AfterViewChecked, ControlValueA
     }
 
     /** @hidden */
-    public ngOnInit(): void {
-        if (!this.nativeElement.placeholder) {
-            this.renderer.setAttribute(this.nativeElement, 'placeholder', this.maskOptions.format);
+    public ngOnChanges(changes: SimpleChanges): void {
```

Comment:
looking further at this we should not use `ngOnChanges`. IMO you should change `mask` to property with getter and setter. Then you can move all the logic from `ngOnChanges` to the mask's setter.

_Note: keeping this in `ngOnChanges` will not work if one calls `inputWithMask.mask = 'some_new_mask'` as this will not fire `ngOnChanges`._